Realm/RLMSyncUtil.h

```diff
@@ -118,6 +121,30 @@ typedef RLM_ERROR_ENUM(NSInteger, RLMSyncError, RLMSyncErrorDomain) {
      error domain.
      */
     RLMSyncErrorUnderlyingAuthError     = 8,
+
+    /**
+     An error that indicates the user does not have permission to perform an operation
+     upon a synced Realm. For example, a user may receive this error if they attempt to
+     open a Realm they do not have at least read access to, or write to a Realm they only
+     have read access to.
+     
+     This error may also occur if a user incorrectly opens a Realm they have read-only
+     permissions to without using the `asyncOpen()` APIs.
+
+     A Realm that suffers a permission denied error is, by default, flagged so that its
+     local copy will be deleted the next time the application starts.
+     
+     The `userInfo` dictionary contains a block under the key
+     `kRLMSyncInitiateDeleteRealmBlockKey`. This block can be called with a single argument:
+     `YES` to immediately delete the Realm file, `NO` to not delete the file at all (either
+     now or upon restart). This block should only be called with `YES` if and when your app
```

Comment:
It seems like there's three possible actions that can be taken:
1. Delete the file now.
2. Delete the file on next launch.
3. Don't delete the file.

But we're using a boolean to select between only two of them?

It's also weird that calling the "initiate delete realm block" with `NO` doesn't initiate a deletion at all.